app/code/Magento/CatalogSearch/Model/Search/FilterMapper/TermDropdownStrategy/StockConditionApplier.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\CatalogSearch\Model\Search\FilterMapper\TermDropdownStrategy;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Select;
+
+/**
+ * Apply stock condition to select.
+ */
+class StockConditionApplier
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection
+    ) {
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * @param string $alias
+     * @param string $stockAlias
+     * @param Select $select
+     *
+     * @return void
+     */
+    public function execute(
+        string $alias,
+        string $stockAlias,
+        Select $select
+    ) {
+        $select->joinLeft(
```

Comment:
first of all - why do we use `left join here`?

Reply:
Fixed.